### PR TITLE
ENGOPS-6641: Add initial additional-information optional parameter

### PR DESCRIPTION
**Pull request by athegaul**
We (on the EngOps side) want to have some form of dynamic naming for the job name that comes from the pull request title. I tried using `this-job-name`, but then looking in the code and the README, this-job-name is the name of the running workflow. It can be changed only as a matrix. 

With an optional property, we want to provide a little bit more information about the running workflow.

For specific gauntlet commands (executed through clops workflow), it creates a PR that needs to be reviewed and merged. For metrics, we are only interested in opened and merged pull requests. The title of the pull request comes in the form of `network-name - number - command`. But, we will only send `network-name` and `command` in the payload. With this additional information in logs, we can apply filtering and create count metrics about opened and closed PRs for commands we are interested in.

If there is another way to do that, please let me know, and we can discuss what needs to be changed.

### CODE REVIEW

File: src/context.ts [hunk 1]
```diff
@@ -53,6 +54,7 @@ export async function fetchContext(
     event: rest,
     workflowRun: mergedWorkflowRunContext,
     jobRun: mergedJobRunContext,
+    ...(additionalInformation && { additionalInformation })
```
Comment (HenryNguyen5) [inline src/context.ts:57]: This is an optional parameter anyway, we should just pass it in.
```suggestion
    additionalInformation 
```

**Comment by HenryNguyen5** [comment]
> For specific gauntlet commands (executed through clops workflow), it creates a PR that needs to be reviewed and merged. For metrics, we are only interested in opened and merged pull requests. The title of the pull request comes in the form of network-name - number - command. But, we will only send network-name and command in the payload. With this additional information in logs, we can apply filtering and create count metrics about opened and closed PRs for commands we are interested in.

Hi @athegaul, nice PR!

- What do you mean by "about opened and merged pull requests"? Do you mean that you want to exclude any metrics that was once part of a PR that is now in a "closed" state? Rather than "Open", or "Merged"? What about "Draft"?
- It would be better to have `additionalInformation` take in a structured JSON payload. That way, you can parse that field in Loki if needed, or external tooling. It opens it up for better automation. 
- With `additionalInformation` being a structured JSON payload, we'll want to enforce some type of schema so that it is extendable by other users rather than just PRs from clops. I recommend adding a minimum of 1 field to act as a discriminator that downstream consumers of the generated loki logs can use, such as `source`, or `id`.

With the following recommendations above, you could have a payload of `additionalInformation` that would look like:
```ts
interface AdditionalInfo {
	source: string
}

interface CLopsAdditionalInfo extends AdditionalInfo {
  source: 'clops'
  network: string
  command: string
}
```

**Review by HenryNguyen5 — CHANGES_REQUESTED** [review]

**Comment by athegaul** [comment]
Hey @HenryNguyen5, thank you for the feedback!

As for the "about opened and merged pull requests" - by this, I mean filtering PRs with specific commands executed. On the dashboard, this will be represented as Stat (count). For example, if we have a PR where the command `N` is executed, and that PR is in review, the dashboard will show this as a number of open pull requests for command `N`. Same for closed PRs. We want to track how long it takes to execute specific commands.

For the JSON payload structure, would it be okay to create `additionalInformation` with `data?: unknown` property so that others can pass the custom payload and don't have to focus on introducing a new interface for the data? Example: 

```javascript
export interface AdditionalInformation {
  source: string,
  data?: unknown
}
```

Since this is a JSON payload, maybe `Object` can be better instead of `unknown`.

**Comment by HenryNguyen5** [comment]
Stale

**Pull request closed without merging**